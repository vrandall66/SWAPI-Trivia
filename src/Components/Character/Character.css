.Character {
  background-color: black;
  margin-bottom: 30px;
  width: 250px;
}

.favoriteIcon {
  width: 35px;
  margin-top: 10px;
}