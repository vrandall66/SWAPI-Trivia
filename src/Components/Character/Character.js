import React from "react";
import "./Character.css";
import "../CharactersContainer/CharactersContainer.css"
import notFavorited from '../../images/not-favorite-ds.svg'
import favorited from '../../images/favorited-ds.svg'

const Character = ({ characterInfo, updateCharacterFavorite }) => {
  return characterInfo.map( (character) => {
    const { name, species, homeworld, films, characterid, favorite } = character
    const favoriteCharacter = character => {
      let {
        name,
        species,
        homeworld,
        films,
        characterid,
        favorite
      } = character;
      favorite = !favorite;
      return { name, species, homeworld, films, characterid, favorite };
    };
    return (
      <div className="Character" key={characterid}>
        <input 
          className='favoriteIcon'
          type='image' 
          alt='favoriteIcon' 
          src={favorite ? favorited : notFavorited} 
          onClick={() => updateCharacterFavorite(favoriteCharacter(character))}
        />
        <h1>{name}</h1>
        <p>{species}</p>
        <p>{homeworld.homeName}</p>
        <p>{homeworld.homePopulation}</p>
        <p>{films}</p>
      </div>
    );
  });
};

export default Character;
