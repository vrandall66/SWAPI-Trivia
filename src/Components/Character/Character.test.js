import React from "react";
import { shallow } from "enzyme";
import Character from "./Character";

describe("Character", () => {
  let wrapper;
  let characterInfo = [
    {
      characterid: 1,
      favorite: false,
      films: [
        "The Empire Strikes Back",
        "Revenge of the Sith",
        "Return of the Jedi",
        "A New Hope",
        "The Force Awakens"
      ],
      homeworld: { homeName: "Tatooine", homePopulation: "200000" },
      name: "Luke Skywalker",
      species: "Human"
    }
  ];
  let updateCharacterFavorite = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Character
        characterInfo={characterInfo}
        updateCharacterFavorite={updateCharacterFavorite}
      />
    );
  });

  it("should match the wrapper", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should iterate through the characterInfo property and create a Character Card for each character", () => {
    
  });
});
