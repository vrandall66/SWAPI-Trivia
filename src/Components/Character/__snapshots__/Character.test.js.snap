// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character should match the wrapper 1`] = `
<div
  className="Character"
  key="1"
>
  <input
    alt="favoriteIcon"
    className="favoriteIcon"
    onClick={[Function]}
    src="not-favorite-ds.svg"
    type="image"
  />
  <h1>
    Name: 
    Luke Skywalker
  </h1>
  <p>
    Species: 
    Human
  </p>
  <p>
    Homeworld: 
    Tatooine
  </p>
  <p>
    Homeworld Population: 
    200000
  </p>
  <FilmsList
    films={
      Array [
        "The Empire Strikes Back",
        "Revenge of the Sith",
        "Return of the Jedi",
        "A New Hope",
        "The Force Awakens",
      ]
    }
  />
</div>
`;
